src/fidesops/main.py

```diff
@@ -40,6 +42,75 @@
         allow_headers=["*"],
     )
 
+
+@app.middleware("http")
+async def dispatch_log_request(request: Request, call_next: Callable) -> Response:
+    fides_source: Optional[str] = request.headers.get("X-Fides-Source")
+    now: datetime = datetime.now(tz=timezone.utc)
+    endpoint = f"{request.method}: {request.url.path}"
+
+    try:
+        response = await call_next(request)
+        # HTTPExceptions are considered a handled err by default so are not thrown here.
+        # Accepted workaround is to inspect status code of response.
+        # More context- https://github.com/tiangolo/fastapi/issues/1840
+        try:
+            if response.status_code >= 400:
+                response.background = BackgroundTask(
+                    prepare_and_log_request,
+                    endpoint,
+                    response.status_code,
+                    now,
+                    fides_source,
+                    "HTTPException",
```

Comment:
error class can be assumed to be `HTTPException` if status code >=400 and it wasn't a thrown exception. In Starlette, only non-HTTP exceptions are thrown